### PR TITLE
feat: share etcd client

### PR DESCRIPTION
**Pull request by spacewander**
Need to change lua-resty-etcd for this feature
For #2899

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by spacewander** [comment]
Required: https://github.com/api7/lua-resty-etcd/pull/100

File: t/node/healthcheck-passive.t [hunk 1]
```diff
@@ -56,7 +56,7 @@ __DATA__
                                     "interval": 100,
                                     "http_failures": 2
                                 }
-                            },
+                            },]] .. [[
```
Comment (membphis) [inline t/node/healthcheck-passive.t:59]: Why do we need to do this? I'm confused.

File: apisix/core/config_etcd.lua [hunk 2]
```diff
@@ -484,6 +484,39 @@ function _M.getkey(self, key)
 end
 
 
+local get_etcd
+do
+    local etcd_cli
+
+    function get_etcd()
+        if etcd_cli == nil then
```
Comment (membphis) [inline apisix/core/config_etcd.lua:492]: code style:

```lua
if etcd_cli then
    return etcd_cli
end

... ...
```

**Review by membphis — COMMENTED** [review]
we can merge this PR after release version `2.2`.

Because we updated the way to connect to etcd, it takes more time to confirm if it works

File: t/node/healthcheck-passive.t (hunk 1 above)
Comment (spacewander) [reply t/node/healthcheck-passive.t:59]: To work around the "too long code snippet" problem.

Comment (membphis) [reply t/node/healthcheck-passive.t:59]: got it, thx your explain

**Review by membphis — APPROVED** [review]
LGTM

**Review by tokers — APPROVED** [review]

**Pull request merged**